server/server.go

```diff
@@ -363,15 +362,62 @@ func (r *rodeServer) ListVersionedResourceOccurrences(ctx context.Context, reque
 		PageSize:  request.PageSize,
 		PageToken: request.PageToken,
 	})
-
 	if err != nil {
 		return nil, createError(log, "error listing occurrences", err)
 	}
 
-	return &pb.ListVersionedResourceOccurrencesResponse{
+	response := &pb.ListVersionedResourceOccurrencesResponse{
 		Occurrences:   allOccurrences.Occurrences,
 		NextPageToken: allOccurrences.NextPageToken,
-	}, nil
+	}
+
+	if request.FetchRelatedNotes {
+		relatedNotes, err := r.fetchRelatedNotes(ctx, log, allOccurrences.Occurrences)
+		if err != nil {
+			return nil, createError(log, "error fetching related notes", err)
+		}
+
+		response.RelatedNotes = relatedNotes
+	}
+
+	return response, nil
+}
+
+func (r *rodeServer) fetchRelatedNotes(ctx context.Context, logger *zap.Logger, occurrences []*grafeas_proto.Occurrence) (map[string]*grafeas_proto.Note, error) {
+	log := logger.Named("fetchRelatedNotes")
+
+	if len(occurrences) == 0 {
+		return nil, nil
+	}
+
+	noteFiltersMap := make(map[string]string)
+	for _, occurrence := range occurrences {
+		if _, ok := noteFiltersMap[occurrence.NoteName]; !ok {
+			noteFiltersMap[occurrence.NoteName] = fmt.Sprintf(`"name" == "%s"`, occurrence.NoteName)
```

Comment:
Nitpick, but I don't think `name` here needs to have quotes around it. It still works because of how we map to the ES query, but it reads like comparing a string literal to another string literal, which isn't what's happening.